File: shoring/compat/src/test/java/com/panopset/compat/test/FileopMockTest.java
package com.panopset.compat.test;

import java.io.IOException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.panopset.compat.Fileop;

public class FileopMockTest {

  @Test
  void testTouch() throws IOException {
    Fileop.touch(null);
    Fileop.delete(FileopTest.tempFile);
    Assertions.assertFalse(FileopTest.tempFile.exists());
    Fileop.touch(FileopTest.tempFile);
    Assertions.assertTrue(FileopTest.tempFile.exists());
    Fileop.delete(FileopTest.deepFile);
    Assertions.assertFalse(FileopTest.deepFile.exists());
    Fileop.touch(FileopTest.deepFile);
    Assertions.assertTrue(FileopTest.deepFile.exists());
  }

}

File: code/shoring/compat/src/test/kotlin/com/panopset/compat/test/FileopMockTest.kt
package com.panopset.compat.test

import com.panopset.compat.Fileop
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class FileopMockTest {
    @Test
    fun testTouch() {
        Fileop.delete(tempFile)
        Assertions.assertFalse(tempFile.exists())
        Fileop.touch(tempFile)
        Assertions.assertTrue(tempFile.exists())
        Fileop.delete(deepFile)
        Assertions.assertFalse(deepFile.exists())
        Fileop.touch(deepFile)
        Assertions.assertTrue(deepFile.exists())
    }
}
